src/AppRun.c

```diff
@@ -25,182 +25,298 @@ THE SOFTWARE.
 
 **************************************************************************/
 
+#include <dirent.h>
+#include <errno.h>
+#include <libgen.h>
+#include <stdarg.h>
+#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
-#include <unistd.h>
-#include <libgen.h>
-#include <dirent.h>
 #include <string.h>
-#include <errno.h>
+#include <unistd.h>
 
-#define die(...)                                    \
-    do {                                            \
-        fprintf(stderr, "Error: " __VA_ARGS__);     \
-        exit(1);                                    \
-    } while(0);
-#define SET_NEW_ENV(str,len,fmt,...)                \
-    format = fmt;                                   \
-    length = strlen(format) + (len);                \
-    char *str = calloc(length, sizeof(char));     \
-    snprintf(str, length, format, __VA_ARGS__);   \
+/** Macro to set environment variables. */
+#define SET_NEW_ENV(str, len, fmt, ...)         \
+    format = fmt;                               \
+    length = strlen(format);                    \
+    char* str = calloc(length, sizeof(char));   \
+    snprintf(str, length, format, __VA_ARGS__);  \
     putenv(str);
```

Comment:
`str` is never `free`d, this line probably causes mem leaks.

Also, it'd probably be better to put this into a scope/`do {...} while` loop like the above macro to keep these vars separate from the callee's stack.